### PR TITLE
Add enum field type

### PR DESCRIPTION
**Pull request by e-davidson**

### CODE REVIEW

File: projects/table-builder/src/lib/components/header-menu/header-menu.component.html [hunk 1]
```diff
@@ -12,9 +12,9 @@
       <input type="hidden" name="key"  [ngModel]="metaData.key" />
       <input type="hidden" name="fieldType"  [ngModel]="metaData.fieldType" />
 
-      <ng-container *ngIf="metaData.additional?.FilterOptions?.FilterableValues?.length > 0; else defaultFilter">
+      <ng-container *ngIf="valueList; else defaultFilter">
         <input type='hidden' name='filterValue' [ngModel]='myFilterValue' />
-        <div *ngFor='let item of metaData.additional.FilterOptions.FilterableValues; index as i'>
+        <div *ngFor="let item of valueList; index as i" >
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/components/header-menu/header-menu.component.html:17]: unused i

File: projects/table-builder/src/lib/components/header-menu/header-menu.component.html [hunk 2]
```diff
@@ -12,9 +12,9 @@
       <input type="hidden" name="key"  [ngModel]="metaData.key" />
       <input type="hidden" name="fieldType"  [ngModel]="metaData.fieldType" />
 
-      <ng-container *ngIf="metaData.additional?.FilterOptions?.FilterableValues?.length > 0; else defaultFilter">
+      <ng-container *ngIf="valueList; else defaultFilter">
         <input type='hidden' name='filterValue' [ngModel]='myFilterValue' />
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/components/header-menu/header-menu.component.html:16]: I guess the myFilterValue is being picked up somewhere?

File: projects/table-builder/src/lib/interfaces/report-def.ts [hunk 3]
```diff
@@ -75,7 +76,8 @@ export interface Additional   {
     header?: Dictionary<string>,
     body?: Dictionary<string>,
     footer?: Dictionary<string>,
-  } 
+  }
+  enumMap?: any;
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/interfaces/report-def.ts:80]: maybe Dictionary<string | number>

File: projects/table-builder/src/lib/components/initialization-component/initialization-component.html [hunk 4]
```diff
@@ -31,6 +31,9 @@
 <ng-template #expression let-element='element' let-metaData='metaData'>
   <span>{{element | func : metaData.transform }}</span>
 </ng-template>
+<ng-template #enum let-element='element' let-metaData='metaData' let-transform='transform'>
+  <span>{{ metaData.additional.enumMap[element[metaData.key]] }}</span>
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/components/initialization-component/initialization-component.html:35]: did you forget to add the func pipe

**Review by NachmanBerkowitz — COMMENTED** [review]
I don't understand why we need an enum map. If the back end anyways has to create the map, let it just be an array of strings.

File: projects/table-builder/src/lib/components/header-menu/header-menu.component.html (hunk 2 above)
Comment (e-davidson) [reply projects/table-builder/src/lib/components/header-menu/header-menu.component.html:16]: correct

File: projects/table-builder/src/lib/interfaces/report-def.ts (hunk 3 above)
Comment (e-davidson) [reply projects/table-builder/src/lib/interfaces/report-def.ts:80]: I'm not sure what you're gaining? an enum should always match a number to a string. (Even if typescript doesn't work that way but for our purposes that's how we're using it)

Comment (NachmanBerkowitz) [reply projects/table-builder/src/lib/interfaces/report-def.ts:80]: the way that we are using it, it should typed as Dictionary<string>

File: projects/table-builder/src/lib/components/initialization-component/initialization-component.html (hunk 4 above)
Comment (e-davidson) [reply projects/table-builder/src/lib/components/initialization-component/initialization-component.html:35]: Not really. why do I need the func pipe?

**Pull request merged**